lib/message_bus.rb

```diff
@@ -503,6 +503,18 @@ def last_id(channel, site_id = nil)
     backend_instance.last_id(encode_channel_name(channel, site_id))
   end
 
+  # Get the ID of the last message published on multiple channels
+  #
+  # @param [Array<String>] channels - array of channels to fetch
+  # @param [String] site_id - the ID of the site by which to filter
+  #
+  # @return [Array<Integer>] the channel-specific IDs of the last message published to each requested channel
+  def last_ids(*channels, site_id: nil)
+    encoded_channel_names = channels.map { |c| encode_channel_name(c, site_id) }
+    ids = backend_instance.last_ids(*encoded_channel_names)
+    channels.zip(ids).to_h
```

Comment:
this can be micro optimized if you avoid allocating the temporary array

Reply:
As in, something like this?

```
def last_ids(*channels, site_id: nil)
    ids = backend_instance.last_ids(*channels.map { |c| encode_channel_name(c, site_id) })
    channels.zip(ids).to_h
end

Reply:
^ that would still be the same. I think Sam meant the array that's the result of `#zip`, which can be avoided with something like:

```rb
channels.each.with_index.with_object({}) do |channel, index, result|
  result[channel] = ids[index]
end
```

…but I don't know how much more performant that is 🙂